### ISSUE TITLE
Implement pydocstyle rules

### ISSUE THREAD

**Issue by charliermarsh** [astral-sh/ruff#274]
It's possible that we could collect the docstrings on the Rust side, then call out to pydocstyle. Or we could just re-implement the logic ourselves in Rust. The goal would be to enable adopting ruff for users of `flake8-docstring`.

See: https://github.com/PyCQA/pydocstyle

**Comment by charliermarsh** [comment]
Starting: #394, #395.

**Comment by matteosantama** [comment]
Wow you blitzed through these! The [rules](https://github.com/charliermarsh/ruff#rules) section of the `README.md` should be updated to reflect the latest coverage

**Comment by charliermarsh** [comment]
Ah yeah, they're enumerated in that huge table (look for the `DXXX` codes), but the whole thing needs to be redone. Maybe for now I'll split that table into sections to make the rule sets clearer?

**Comment by matteosantama** [comment]
Oh, no that's my bad! I thought the check mark meant "implemented" and the `DXXX` codes needed a check.

**Comment by matteosantama** [comment]
As a follow-up though, is the best way to activate `pycodestyle` to manually enable those error codes? Or is there an "on-switch" for the plugin as a whole?

**Comment by charliermarsh** [comment]
Oh that's great feedback... If the check mark confused you, then it's my bad, not yours :) Will think on that.

**Comment by charliermarsh** [comment]
Yeah, the best way to activate (for now) is to manually enable the error code. I'm going to improve this.

Here's an example from Polars that I was (coincidentally) just writing up: https://github.com/pola-rs/polars/pull/5151#issuecomment-1279765121
